### PR TITLE
patch(TPS-3243): Enable multiple versions of Databricks / Spark + Support ADLS Gen2 for Databricks 5.4 (TBD-7856)

### PR DESCRIPTION
**Pull request by sponomarova**
…port ADLS Gen2 for Databricks 5.4 (TBD-7856)

**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution.dbr350/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-Bundle-Vendor: .Talend SA.
-Service-Component: OSGI-INF/databricks.xml
-Bundle-ActivationPolicy: lazy
-Require-Bundle: org.talend.core.runtime,
- org.talend.hadoop.distribution
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Databricks
+Bundle-SymbolicName: org.talend.hadoop.distribution.dbr350;singleton:=true
+Bundle-Version: 7.3.1.qualifier
```
Comment (ikulish) [inline main/plugins/org.talend.hadoop.distribution.dbr350/META-INF/MANIFEST.MF]: Is it ok to change to 7.3.1 for 7.2 patch?

File: test/features/org.talend.bigdata.se.test.feature/feature.xml [hunk 2]
```diff
@@ -236,13 +236,23 @@
...
+         id="org.talend.hadoop.distribution.dbr540.test"
+         download-size="0"
+         install-size="0"
+         version="0.0.0"
+         fragment="true"
+         unpack="false"/>
+
    <plugin
          id="org.talend.hadoop.distribution.qubole.test"
          download-size="0"
```
Comment (ikulish) [inline test/features/org.talend.bigdata.se.test.feature/feature.xml]: Why there is no change for main/features/org.talend.hadoop.distribution.feature/feature.xml from https://github.com/Talend/tbd-studio-se/pull/1061?

File: main/plugins/org.talend.hadoop.distribution.dbr350/pom.xml [hunk 3]
```diff
@@ -4,9 +4,9 @@
   <parent>
     <groupId>org.talend.studio</groupId>
     <artifactId>tbd-studio-se</artifactId>
-    <version>7.2.1</version>
+    <version>7.3.1-SNAPSHOT</version>
```
Comment (ikulish) [inline main/plugins/org.talend.hadoop.distribution.dbr350/pom.xml]: Is it correct to change studio version for the patch?

File: main/plugins/org.talend.hadoop.distribution.dbr540/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Dbr540
+Bundle-SymbolicName: org.talend.hadoop.distribution.dbr540;singleton:=true
+Bundle-Version: 7.3.1.qualifier
```
Comment (ikulish) [inline main/plugins/org.talend.hadoop.distribution.dbr540/META-INF/MANIFEST.MF]: is it correct version?

File: main/plugins/org.talend.hadoop.distribution.dbr540/pom.xml [hunk 5]
```diff
@@ -4,9 +4,9 @@
   <parent>
     <groupId>org.talend.studio</groupId>
     <artifactId>tbd-studio-se</artifactId>
-    <version>7.2.1</version>
+    <version>7.3.1-SNAPSHOT</version>
```
Comment (ikulish) [inline main/plugins/org.talend.hadoop.distribution.dbr540/pom.xml]: is it correct version?

File: test/features/org.talend.bigdata.se.test.feature/feature.xml (hunk 2 above)
Comment (ikulish) [inline test/features/org.talend.bigdata.se.test.feature/feature.xml]: Why there is no change for main/features/org.talend.hadoop.distribution.feature/feature.xml from https://github.com/Talend/tbd-studio-se/pull/1061?

File: test/features/org.talend.bigdata.se.test.feature/pom.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.talend.studio</groupId>
     <artifactId>tbd-studio-se</artifactId>
-    <version>7.2.1</version>
+    <version>7.3.1-SNAPSHOT</version>
```
Comment (ikulish) [inline test/features/org.talend.bigdata.se.test.feature/pom.xml]: is it correct version?

File: main/plugins/org.talend.hadoop.distribution.dbr540/META-INF/MANIFEST.MF (hunk 4 above)
Comment (ikulish) [inline main/plugins/org.talend.hadoop.distribution.dbr540/META-INF/MANIFEST.MF]: is it correct version?

File: main/plugins/org.talend.hadoop.distribution.dbr350/pom.xml (hunk 3 above)
Comment (ikulish) [inline main/plugins/org.talend.hadoop.distribution.dbr350/pom.xml]: is it correct version?

File: test/features/org.talend.bigdata.se.test.feature/feature.xml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 <feature
       id="org.talend.bigdata.se.test.feature"
       label="BigData SE Test Feature"
-      version="7.2.1.qualifier">
+      version="7.3.1.qualifier">
```
Comment (ikulish) [inline test/features/org.talend.bigdata.se.test.feature/feature.xml]: is it correct version?

File: test/plugins/org.talend.hadoop.distribution.dbr350.test/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -1,10 +1,10 @@
 Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
-Bundle-Name: Azure Databricks Distribution Test
-Bundle-SymbolicName: org.talend.hadoop.distribution.databricks.test
-Bundle-Version: 7.2.1.qualifier
+Bundle-Name: Databricks 3.5.0 Distribution Test
+Bundle-SymbolicName: org.talend.hadoop.distribution.dbr350.test
+Bundle-Version: 7.3.1.qualifier
```
Comment (ikulish) [inline test/plugins/org.talend.hadoop.distribution.dbr350.test/META-INF/MANIFEST.MF]: is it correct version?

File: test/plugins/org.talend.hadoop.distribution.dbr350.test/pom.xml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.talend.studio</groupId>
+    <artifactId>tbd-studio-se</artifactId>
+    <version>7.3.1-SNAPSHOT</version>
```
Comment (ikulish) [inline test/plugins/org.talend.hadoop.distribution.dbr350.test/pom.xml]: the same question

File: test/plugins/org.talend.hadoop.distribution.dbr540.test/META-INF/MANIFEST.MF [hunk 10]
```diff
@@ -0,0 +1,11 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Databricks 5.4.0 Distribution Test
+Bundle-SymbolicName: org.talend.hadoop.distribution.dbr540.test
+Bundle-Version: 7.3.1.qualifier
```
Comment (ikulish) [inline test/plugins/org.talend.hadoop.distribution.dbr540.test/META-INF/MANIFEST.MF]: the same question

File: test/plugins/org.talend.hadoop.distribution.dbr540.test/pom.xml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.talend.studio</groupId>
+    <artifactId>tbd-studio-se</artifactId>
+    <version>7.3.1-SNAPSHOT</version>
```
Comment (ikulish) [inline test/plugins/org.talend.hadoop.distribution.dbr540.test/pom.xml]: the same question

File: main/plugins/org.talend.hadoop.distribution.dbr350/pom.xml (hunk 3 above)
Comment (sponomarova) [reply main/plugins/org.talend.hadoop.distribution.dbr350/pom.xml]: Corrected.

File: main/plugins/org.talend.hadoop.distribution.dbr350/META-INF/MANIFEST.MF (hunk 1 above)
Comment (sponomarova) [reply main/plugins/org.talend.hadoop.distribution.dbr350/META-INF/MANIFEST.MF]: Corrected.

File: main/plugins/org.talend.hadoop.distribution.dbr540/META-INF/MANIFEST.MF (hunk 4 above)
Comment (sponomarova) [reply main/plugins/org.talend.hadoop.distribution.dbr540/META-INF/MANIFEST.MF]: Corrected.

File: main/plugins/org.talend.hadoop.distribution.dbr540/pom.xml (hunk 5 above)
Comment (sponomarova) [reply main/plugins/org.talend.hadoop.distribution.dbr540/pom.xml]: No. It isn't. Modified to 7.2.

File: test/features/org.talend.bigdata.se.test.feature/feature.xml (hunk 7 above)
Comment (sponomarova) [reply test/features/org.talend.bigdata.se.test.feature/feature.xml]: No. It isn't. Modified to 7.2.

File: test/plugins/org.talend.hadoop.distribution.dbr350.test/pom.xml (hunk 9 above)
Comment (sponomarova) [reply test/plugins/org.talend.hadoop.distribution.dbr350.test/pom.xml]: the same answer :)

File: test/features/org.talend.bigdata.se.test.feature/feature.xml [hunk 12]
```diff
@@ -19,7 +19,7 @@
    <plugin
          id="org.talend.hadoop.distribution.test"
          download-size="0"
-         install-size="0"
+         install-size="0"l
```
[the remark this answers is not in the document]
Comment (sponomarova) [reply test/features/org.talend.bigdata.se.test.feature/feature.xml]: yes. Will correct that. Thanks!

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractDatabricksDistribution.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+//
+// ============================================================================
+package org.talend.hadoop.distribution;
+
+public abstract class AbstractDatabricksDistribution extends AbstractDistribution {
+
+    /**
+     * Get the spark version to be passed in rest call for transient cluster creation
+     */
+    public String getTransientClusterSparkVersion() {
```
Comment (AlixMetivier) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractDatabricksDistribution.java]: shouldn't this be used in spark footer or header ? Or is the hard coded version enough ? i did not see any file changed about this

**Comment by amedvedenko** [comment]
LGTM

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractDatabricksDistribution.java (hunk 13 above)
Comment (sponomarova) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractDatabricksDistribution.java]: @AlixMetivier thanks for taking a look.
Probably this point may be discussed in scope of fix on master.
However it is PR for patch branch I believe it should be consistent with changes on master.

File: test/features/org.talend.bigdata.se.test.feature/pom.xml (hunk 6 above)
Comment (aparent-talend) [reply test/features/org.talend.bigdata.se.test.feature/pom.xml]: it's not

Comment (aparent-talend) [reply test/features/org.talend.bigdata.se.test.feature/pom.xml]: Needs to be changed

**Pull request merged**